data/model/proxy_cache.py

```diff
@@ -43,3 +43,21 @@ def get_proxy_cache_config_for_org(org_name):
         )
     except ProxyCacheConfig.DoesNotExist as e:
         raise InvalidProxyCacheConfigException(str(e))
+
+
+def delete_proxy_cache_config(org_name):
```

Comment:
This can be accomplished with a single query:
```python
    try:
        ProxyCacheConfig.select().join(User).where(
            (User.username == org_name) & (User.organization == True)
        ).get().delete_instance()
        return True
    except ProxyCacheConfig.DoesNotExist:
        pass
    return False
```
But we can take that as a follow-up.